vscode/resources/sample-provider-settings.yaml

```diff
@@ -0,0 +1,40 @@
+---
+# This is a sample settings file with a few common providers and models defined
+
+environment:
+  ALWAYS_APPLIED_KEY: "envvar to be set regardless of which model is active"
+
+models:
+  OpenAI: &active
+    environment:
+      OPENAI_API_KEY: "key"
+    provider: "ChatOpenAI"
+    args:
+      model: "gpt-4"
+
+  AmazonBedrock:
+    environment:
+      AWS_ACCESS_KEY_ID: "key id"
+      AWS_SECRET_ACCESS_KEY: "secret"
```

Comment:
The file contains both public info that could be part of the repository (i.e. model name) and secrets that should be kept away from workspace and repository. Perhaps we should separate both and merge them in the model provider?

Reply:
Lots of variations can be done.

This I'm working off my meeting notes here: https://github.com/konveyor/editor-extensions/issues/184#issuecomment-2583077916

At least for now, I understood the goals as:
  - have one place to configure things so it is easy to configure
  - keep that place outside of the repo so sensitive info won't get committed by accident
  - bonus points if that place can have comments and examples